SIPS/sip-23.md

```diff
@@ -0,0 +1,599 @@
+
+```typescript jsx
+<Button onClick={handleClick}>Click me</Button>
+```
+
+#### Container
+
+The `Container` component renders a container element. It accepts a `children`
+prop which can be a tuple of `BoxElement` and `FooterElement` or a single
+`BoxElement`.
+
+```typescript
+type ContainerProps = {
+  children: [
+    BoxElement,
+    FooterElement
+  ] | [
+    BoxElement
+  ]
+};
+
+const Container: SnapComponent<ContainerProps>;
+```
+
+##### Example
+
+```typescript jsx
+<Container>
+  <Box>
+    <Text>Hello, world!</Text>
+  </Box>
+  <Footer>
+    <Button onClick={handleClick}>Click me</Button>
+  </Footer>
+</Container>
+```
+
+#### Copyable
+
+The `Copyable` component renders a text element with a copy button. It accepts a
+`value` prop which is a string to be copied to the clipboard when the button is
+clicked. The component also accepts an optional `sensitive` prop which can be
+used to indicate that the value is sensitive and should be obscured when
+displayed.
+
+```typescript
+type CopyableProps = {
+  value: string;
+  sensitive?: boolean | undefined;
+};
+
+const Copyable: SnapComponent<CopyableProps>;
+```
+
+##### Example
+
+```typescript jsx
+<Copyable value="0x4bbeEB066eD09B7AEd07bF39EEe0460DFa261520" />
+<Copyable value="my-secret-key" sensitive />
+```
+
+#### Divider
+
+The `Divider` component renders a horizontal line. It does not accept any props.
+
+```typescript
+const Divider: SnapComponent;
+```
+
+##### Example
+
+```typescript jsx
+<Divider />
+```
+
+#### Field
+
+The `Field` component renders a field element. It accepts a `label` prop which
+is a string representing the field label, and a `children` prop which can be an
+input element, and an optional button element. The component also accepts an
+`error` prop which is a string representing an error message.
+
+```typescript
+type FieldProps = {
+  label: string;
+  error?: string | undefined;
+  children: [InputElement, ButtonElement] | InputElement;
+};
+```
+
+##### Example
+
+```typescript jsx
+<Field label="Name">
+  <Input type="text" />
+  <Button onClick={handleClick}>Action</Button>
+</Field>
+```
+
+#### Footer
+
+The `Footer` component renders a footer element. It accepts a `children` prop
+which can be one or two buttons.
+
+```typescript
+type FooterProps = {
+  children: ButtonElement | [ButtonElement, ButtonElement];
+};
+
+const Footer: SnapComponent<FooterProps>;
+```
+
+##### Example
+
+```typescript jsx
+<Footer>
+  <Button onClick={handleCancel}>Cancel</Button>
+  <Button onClick={handleConfirm}>Confirm</Button>
+</Footer>
+
+<Footer>
+  <Button onClick={handleAction}>Action</Button>
+</Footer>
+```
+
+#### Form
+
+The `Form` component renders a form element. It accepts a `children` prop which
+can be one or more field elements.
+
+```typescript
+type FormProps = {
+  children: FieldElement | FieldElement[];
+};
+```
+
+##### Example
+
+```typescript jsx
+<Form>
+  <Field label="Name">
+    <Input type="text" />
+  </Field>
+</Form>
+```
```

Comment:
Same here ?

Reply:
And do we not need a `onSubmit` ?

Reply:
I forgot about `onSubmit` 😅 Will update.